src/utils/preRestart.ts

```diff
@@ -1,28 +1,41 @@
 /* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { Signer } from 'ethers'
 import { Route } from '../types'
 import { LifiErrorCode } from './errors'
 
-export const handlePreRestart = (route: Route) => {
+export const handlePreRestart = async (route: Route, signer: Signer) => {
   for (let index = 0; index < route.steps.length; index++) {
     const stepHasFailed = route.steps[index].execution?.status === 'FAILED'
 
     if (stepHasFailed) {
-      handleErrorType(route, index)
+      await handleErrorType(route, index, signer)
       deleteFailedProcesses(route, index)
       deleteTransactionData(route, index)
     }
   }
 }
 
-const handleErrorType = (route: Route, index: number) => {
+const handleErrorType = async (route: Route, index: number, signer: Signer) => {
   const isGasLimitError = route.steps[index].execution?.process.some(
     (p) => p.error?.code === LifiErrorCode.GasLimitError
   )
   const isGasPriceError = route.steps[index].execution?.process.some(
     (p) => p.error?.code === LifiErrorCode.TransactionUnderpriced
   )
 
+  const { transactionRequest } = route.steps[index]
+
   if (isGasLimitError) {
+    if (transactionRequest) {
+      let gasLimit = transactionRequest.gasLimit
+
+      try {
+        gasLimit = await signer.estimateGas(transactionRequest)
+      } catch (error) {}
```

Comment:
I recently had a conversation with @chybisov where he mentioned that this is best being ignored.

although, at that time I'd mistakenly written `console.log` instead of `console.error`.

I'd still prefer having an error being logged. because the rest of the error are being handled in custom error handlers.

wdyt?
cc: @chybisov

Reply:
We should have a fallback in case it throws, but we don't log anything in the SDK now, so I see no reason to log this one for now, but maybe we should come up with something in the future. We should especially avoid `console.error` as it can annoy some envs.

Reply:
alright! We should make sure that a failure here does not break the process. We should see that estimation as an addition. If we don't set anything, metamask sets its own estimations.
So it's sort of a nice to have if it works but not breaking and falling back to metamask defaults if it does not work. It looks like the code does exactly that.

Reply:
jup true